projects/24_push_constants/PushConstants.cpp

```diff
@@ -0,0 +1,369 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "PushConstants.h"
+
+#include "ppx/graphics_util.h"
+using namespace ppx;
+
+#if defined(USE_DX12)
+const grfx::Api kApi = grfx::API_DX_12_0;
+#elif defined(USE_VK)
+const grfx::Api kApi = grfx::API_VK_1_1;
+#endif
+
+void PushConstantsApp::Config(ppx::ApplicationSettings& settings)
+{
+    settings.appName                    = "24_push_constants";
+    settings.enableImGui                = true;
+    settings.grfx.api                   = kApi;
+    settings.grfx.swapchain.depthFormat = grfx::FORMAT_D32_FLOAT;
+    settings.grfx.enableDebug           = true;
```

Comment:
seems like we ship samples with debug = false by default.

Reply:
Oops, my bad, I'll turn it off. Had it on to debug during dev.